src/index.ts

```diff
@@ -15,34 +17,107 @@ const client = new Client({
   database: process.env.DB,
 });
 
-client
-  .connect()
+client.connect()
   .then(() => {
     console.log("Connected to PostgreSQL database");
+    client.query("SELECT * FROM users", (err, result) => {
+      if (err) {
+        console.error("Error executing query", err.stack);
+      } else {
+        console.log("Query result:", result.rows);
+      }
+    });
   })
   .catch((err) => {
-    console.error("Error connecting to PostgreSQL database", err);
+    console.error("Error connecting to PostgreSQL database", err.stack);
   });
 
-client.query("SELECT * FROM users", (err, result) => {
-  if (err) {
-    console.error("Error executing query", err);
-  } else {
-    console.log("Query result:", result.rows);
-  }
-});
-
 app.use(express.urlencoded({ extended: true }));
 app.use(express.json());
 app.use(pagination);
 
+// Endpoint לבדיקה כללית
 app.get("/smart", (_, res: Response) => {
   res.status(200).json({
     success: true,
     message: "You are on node-typescript-boilerplate.",
   });
 });
 
+
+// Endpoint לכניסה למערכת
+app.post("/login", (req: Request, res: Response) => {
+  const { username, password } = req.body;
+
+  if (!username || !password) {
+    return res.status(400).json({ success: false, message: "Username and password are required" });
+  }
+
+  const query = `SELECT id, username, password, role FROM public.users WHERE username = $1`;
+  
+  client.query(query, [username], (err, result) => {
+    if (err) {
+      console.error("Error executing query:", err.stack);
+      return res.status(500).json({ success: false, message: "Database error" });
+    }
+
+    const user = result.rows[0];
+    if (!user) {
+      return res.status(401).json({ success: false, message: "Invalid username or password" });
+    }
+
+    bcrypt.compare(password, user.password, (err, match) => {
+      if (err || !match) {
+        return res.status(401).json({ success: false, message: "Invalid username or password" });
+      }
+
+      const token = jwt.sign({ id: user.id, username: user.username, role: user.role }, process.env.JWT_SECRET!, { expiresIn: '1h' });
+      res.status(200).json({ success: true, message: "Login successful", token });
+    });
+  });
+});
+
+
+app.post("/register", async (req: Request, res: Response) => {
+  const { username, password, role } = req.body;
+
+  if (!username || !password || !role) {
+    return res.status(400).json({ success: false, message: "Username, password, and role are required" });
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
```

Comment:
This is not how it should be done. You should hash the password with a salt and append it to the end, then save it to the database.